pinecone/deployment/src/test/java/io/quarkiverse/langchain4j/pinecone/deployment/PineconeEmbeddingStoreTest.java

```diff
@@ -0,0 +1,298 @@
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.CosineSimilarity;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.RelevanceScore;
+import io.quarkiverse.langchain4j.pinecone.PineconeEmbeddingStore;
+import io.quarkiverse.langchain4j.pinecone.runtime.DeleteRequest;
+import io.quarkiverse.langchain4j.pinecone.runtime.PineconeVectorOperationsApi;
+import io.quarkiverse.langchain4j.pinecone.runtime.QueryRequest;
+import io.quarkiverse.langchain4j.pinecone.runtime.VectorMatch;
+import io.quarkus.logging.Log;
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Prerequisites for this test: A pinecone index must exist (can be in the starter region)
+ * and the following environment variables must be set accordingly:
+ * PINECONE_API_KEY, PINECONE_ENVIRONMENT, PINECONE_PROJECT_ID and PINECONE_INDEX_NAME
+ *
+ * These are set as GitHub secrets in the main repository. GitHub doesn't
+ * pass them to workflows triggered from forks though, so this test only
+ * runs with the nightly CI workflow, or for PRs submitted from the main
+ * quarkiverse repository (NOT from a fork).
+ *
+ * <p>
+ * Original data in the index will be lost during the test.
+ * <p>
+ * Because of delays in Pinecone when deleting vectors, the test adds
+ * artificial delays (the `delay` method) to make sure we see the correct
+ * data, and thus the test takes a relatively long time to run. If you see
+ * intermittent failures, it may mean that the delay isn't long enough...
+ *
+ */
+@EnabledIfEnvironmentVariable(named = "PINECONE_API_KEY", matches = ".+")
+public class PineconeEmbeddingStoreTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource(new StringAsset(
+                            // to enable rest client logging:
+                            "quarkus.rest-client.logging.scope=request-response\n" +
+                                    "quarkus.rest-client.logging.body-limit=10000\n" +
+                                    "quarkus.log.category.\"org.jboss.resteasy.reactive.client.logging\".level=DEBUG\n" +
+                                    "quarkus.langchain4j.pinecone.api-key=${pinecone.api.key}\n" +
+                                    "quarkus.langchain4j.pinecone.environment=${pinecone.environment}\n" +
+                                    "quarkus.langchain4j.pinecone.project-id=${pinecone.project-id}\n" +
+                                    "quarkus.langchain4j.pinecone.index-name=${pinecone.index-name}\n"),
+                            "application.properties"));
+
+    @Inject
+    PineconeEmbeddingStore embeddingStore;
```

Comment:
Does `@Inject EmbeddingStore` work too?

Reply:
Right now, `@Inject EmbeddingStore` works, but if https://github.com/quarkiverse/quarkus-langchain4j/pull/73 is OK I will update also it here to accept `@Inject EmbeddingStore<TextSegment>`, it's better than using the raw type